qiskit/transpiler/passes/optimization/optimize_swap_before_measure.py

```diff
@@ -35,29 +35,33 @@ def run(self, dag):
         Returns:
             DAGCircuit: the optimized DAG.
         """
-        swaps = dag.op_nodes(SwapGate)
-        for swap in swaps[::-1]:
-            final_successor = []
-            for successor in dag.successors(swap):
-                final_successor.append(successor.type == 'out' or (successor.type == 'op' and
-                                                                   successor.op.name == 'measure'))
-            if all(final_successor):
-                # the node swap needs to be removed and, if a measure follows, needs to be adapted
-                swap_qargs = swap.qargs
-                measure_layer = DAGCircuit()
-                for qreg in dag.qregs.values():
-                    measure_layer.add_qreg(qreg)
-                for creg in dag.cregs.values():
-                    measure_layer.add_creg(creg)
-                for successor in list(dag.successors(swap)):
-                    if successor.type == 'op' and successor.op.name == 'measure':
-                        # replace measure node with a new one, where qargs is set with the "other"
-                        # swap qarg.
-                        dag.remove_op_node(successor)
-                        old_measure_qarg = successor.qargs[0]
-                        new_measure_qarg = swap_qargs[swap_qargs.index(old_measure_qarg) - 1]
-                        measure_layer.apply_operation_back(Measure(), [new_measure_qarg],
-                                                           [successor.cargs[0]])
-                dag.compose(measure_layer)
-                dag.remove_op_node(swap)
-        return dag
+
+        new_dag = DAGCircuit()
+        new_dag.metadata = dag.metadata
+        new_dag._global_phase = dag._global_phase
+        for creg in dag.cregs.values():
+            new_dag.add_creg(creg)
+        for qreg in dag.qregs.values():
+            new_dag.add_qreg(qreg)
```

Comment:
```suggestion
        new_dag = dag._copy_circuit_metadata()
```

Reply:
nice one. Why is `_copy_circuit_metadata` a private method?

Reply:
Initially because it was used internally, but then was useful to passes that needed to rebuild the dag (but preserve metadata) like the layout routing passes. Making it public seems like a good idea.